### PR TITLE
add create user and delete user feature

### LINKED ISSUE #2374: [enhancement] Add / Delete Users by Admin Users
Currently there is no add / delete users for user management under admin. If nobody has any concern I can do a PR for this.

### PR DESCRIPTION
**Pull request by sendilkumarn**
Admin - Create User : 
when a new user is created, reset key and date will be generated and mail will be triggered. Users upon clicking the link, system will ask them to reset the password. By default, Users created via backend will always be activated.
Admin - Delete User :
User will be hard deleted.

fix #2374

### CODE REVIEW

File: app/templates/src/main/java/package/service/_UserService.java [hunk 1]
```diff
@@ -117,7 +118,17 @@ public User createUserInformation(String login, String password, String firstNam
         log.debug("Created Information for User: {}", newUser);
         return newUser;
     }
-
+    public User createUser(User user) {
+        String encryptedPassword = passwordEncoder.encode(RandomUtil.generatePassword());
+        user.setPassword(encryptedPassword);
+        user.setResetKey(RandomUtil.generateResetKey());
+        user.setResetDate(ZonedDateTime.now());
```
Comment (atomfrede) [inline app/templates/src/main/java/package/service/_UserService.java]: I think you need to differentiate here between mongo/sql and cassandra. Cassandra requires a plain ``java.util.Date``

Comment (sendilkumarn) [inline app/templates/src/main/java/package/service/_UserService.java]: Thanks @atomfrede

**Comment by deepu105** [comment]
I tested this for SQL and looks good, if everyone is ok I'll merge this

**Comment by atomfrede** [comment]
Is there also a possibility to just (temporarly) deactivate the user?

**Comment by sendilkumarn** [comment]
No this feature includes only deletion of user.

Temporary deactivation can be done via edit screen right?

**Comment by atomfrede** [comment]
Yes, I was not sure if it is implemented there.

**Comment by sendilkumarn** [comment]
Yes it is available in both listing screen and edit screen.

File: app/templates/src/main/webapp/scripts/app/admin/user-management/_user-management-delete-dialog.controller.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+'use strict';
+
+angular.module('<%=angularAppName%>')
+	.controller('user-managementDeleteController', function($scope, $modalInstance, entity, User) {
```
Comment (deepu105) [inline app/templates/src/main/webapp/scripts/app/admin/user-management/_user-management-delete-dialog.controller.js]: #2389 is merged hence plz update to uibModalInstance

File: app/templates/src/main/webapp/scripts/app/admin/user-management/_user-management.js [hunk 3]
```diff
@@ -93,5 +93,28 @@ angular.module('<%=angularAppName%>')
...
                 }]
+            })
+            .state('user-management.delete', {
+                parent: 'user-management',
+                url: '/{login}/delete',
+                data: {
+                    authorities: ['ROLE_ADMIN'],
+                },
+                onEnter: ['$stateParams', '$state', '$modal', function($stateParams, $state, $modal) {
+                    $modal.open({
```
Comment (deepu105) [inline app/templates/src/main/webapp/scripts/app/admin/user-management/_user-management.js]: #2389 merged plz chnage to uibModal

**Comment by jdubois** [comment]
Great feature, I just tried it and I have a few issues to be solved before we can merge this:

- The "create a new user" button is not at the right place -> can you use the same layout as what we have with the entity generator?
- If we use an existing email address, and error will occur -> the error is not displayed correctly on the screen, there is just an alert box with no text, and it is not cleanly displayed
- When creating a new user, even if I select the "ROLE_USER" authorization, the user does not have that authorization. I need to edit the user, add the authorization again, and then it works.
- Deleting a user does not work, I have a `ReferenceError: $uibModal is not defined` error in my browser -> that might come from some of our recent changes, not your fault

**Comment by sendilkumarn** [comment]
@jdubois  
>for If we use an existing email address, and error will occur -> the error is not displayed correctly on the >screen, there is just an alert box with no text, and it is not cleanly displayed

 `ResponseEntity.badRequest().headers("alert" "param")).body(null); `

always throws empty alert. Try simulating the same in any generated entity. I would suggest we can write a custom header util for this kind of custom messages.

**Comment by deepu105** [comment]
I guess its ok to write additional HeaderUtil methods for such validation errors

**Comment by sendilkumarn** [comment]
@jdubois resolved all the issues listed.

**Comment by sendilkumarn** [comment]
@jdubois  Please let me know if you have any more comments.

**Comment by jdubois** [comment]
Great work!!
I have also tested this with Cassandra and MongoDB

I am going to merge this ASAP:

- I'm waiting for Travis CI to do all the automated checks
- I'm going to rename some translation keys as they are not correct (but this isn't your fault!)

**Comment by jdubois** [comment]
I have merged this manually so I'm closing this -> strange that Github does not show this as "merged", if you see any error please ping me!

**Pull request closed without merging**